numpy/core/tests/test_shape_base.py

```diff
@@ -375,39 +376,30 @@ def test_stack():
                         stack, [np.arange(2), np.arange(3)])
 
 
-# See for more information on how to parametrize a whole class
-# https://docs.pytest.org/en/latest/example/parametrize.html#parametrizing-test-methods-through-per-class-configuration
-def pytest_generate_tests(metafunc):
-    # called once per each test function
-    if hasattr(metafunc.cls, 'params'):
-        arglist = metafunc.cls.params
-        argnames = sorted(arglist[0])
-        metafunc.parametrize(argnames,
-                             [[funcargs[name] for name in argnames]
-                              for funcargs in arglist])
-
-
-# blocking small arrays and large arrays go through different paths.
-# the algorithm is triggered depending on the number of element
-# copies required.
-# We define a test fixture that forces most tests to go through
-# both code paths.
-# Ultimately, this should be removed if a single algorithm is found
-# to be faster for both small and large arrays.s
-def _block_force_concatenate(arrays):
-    arrays, list_ndim, result_ndim, _ = _block_setup(arrays)
-    return _block_concatenate(arrays, list_ndim, result_ndim)
-
-
-def _block_force_slicing(arrays):
-    arrays, list_ndim, result_ndim, _ = _block_setup(arrays)
-    return _block_slicing(arrays, list_ndim, result_ndim)
-
-
 class TestBlock(object):
-    params = [dict(block=block),
-              dict(block=_block_force_concatenate),
-              dict(block=_block_force_slicing)]
+    @pytest.fixture(params=['block', 'force_concatenate', 'force_slicing'])
+    def block(request): # maybe needs self, not sure
```

Comment:
Assuming this works, should strip this comment out

Reply:
man, what a mistake!